implementations/rust/ockam/ockam_api/src/authenticator/direct.rs

```diff
@@ -0,0 +1,419 @@
+pub mod types;
+
+use crate::auth::types::Attributes;
+use crate::signer::types::{Credential, IdentityId};
+use crate::util::response;
+use crate::{assert_request_match, assert_response_match, Cbor};
+use crate::{signer, Timestamp};
+use crate::{Error, Method, Request, RequestBuilder, Response, ResponseBuilder, Status};
+use core::time::Duration;
+use core::{fmt, str};
+use minicbor::encode::write::Cursor;
+use minicbor::{Decoder, Encode};
+use ockam_core::errcode::{Kind, Origin};
+use ockam_core::{self, Address, Result, Route, Routed, Worker};
+use ockam_identity::authenticated_storage::AuthenticatedStorage;
+use ockam_identity::{IdentityIdentifier, IdentitySecureChannelLocalInfo};
+use ockam_node::Context;
+use tracing::{trace, warn};
+use types::{AddEnroller, AddMember, Placeholder};
+
+// storage scopes:
+const ENROLLER: &str = "enroller";
+const MEMBER: &str = "member";
+
+const MAX_VALIDITY: Duration = Duration::from_secs(2 * 3600);
+
+#[derive(Debug)]
+pub struct Server<S, T> {
+    m_store: S, // member store
```

Comment:
It used to be one, but after discussions with @polvorin it was changed so that enrollers live in RAM only and are set when the project node is provisioned. With the current protocol, member information needs to be persistent, which lead to the separation of the stores.

Reply:
I guess that means that list of enrollers live somewhere on elixir side? This is very weird.